Brutus/src/compiler/codegen.jl

```diff
@@ -0,0 +1,282 @@
+#####
+##### Codegen
+#####
+
+# This is the Julia interface between Julia's IRCode and JLIR.
+
+function maybe_widen_type(b::JLIRBuilder, loc::JLIR.Location, 
+        jlir_value::JLIR.Value, expected_type::Type)
+    jlir_type = JLIR.get_type(jlir_value)
+    type = convert_jlirtype_to_type(jlir_type)
+    if (type != expected_type && type <: expected_type)
+        jlir_expected_type = convert_type_to_jlirtype(b.ctx, expected_type)
+        op = create!(b, PiOp(), loc, jlir_value, jlir_expected_type)
+        return JLIR.get_result(op, 0)
+    else
+        return jlir_value
+    end
+end
+
+function emit_value(b::JLIRBuilder, loc::JLIR.Location, 
+        value, ::Type)
+    type = typeof(value)
+    jlir_type = convert_type_to_jlirtype(b.ctx, type)
```

Comment:
Instead of snake case, either `convert(JLIRType, type, b.ctx)` or `JLIRType(b.ctx, type)`